packages/renderers-vixen-parser/e2e/generate-anchor.cjs

```diff
@@ -16,24 +16,44 @@ function transformHyphensToUnderscores(input) {
 async function main() {
     const project = process.argv.slice(2)[0] ?? undefined;
     const sdkName = process.argv.slice(3)[0] ? transformHyphensToUnderscores(process.argv.slice(3)[0]) : undefined;
+    const generateProto = process.argv.slice(4)[0] === 'true' ? true : false;
+
     if (project === undefined) {
         throw new Error('Project name is required.');
     }
     if (sdkName === undefined) {
         throw new Error('SDK name is required.');
     }
-    await generateProject(project, sdkName);
+
+    // create src/generated directory if it doesn't exist
+    const generatedDir = path.join(__dirname, project, 'parser', 'src', 'generated');
+    if (!fs.existsSync(generatedDir)) {
+        fs.mkdirSync(generatedDir, { recursive: true });
+    }
+
+    // create proto directory if it doesn't exist
+    const protoDir = path.join(__dirname, project, 'parser', 'proto');
+    if (!fs.existsSync(protoDir)) {
+        fs.mkdirSync(protoDir);
+    }
+
+    await generateProject(project, sdkName, generateProto);
 }
 
-async function generateProject(project, sdkName) {
+async function generateProject(project, sdkName, generateProto) {
     const idl = readJson(path.join(__dirname, project, 'idl.json'));
     const node = rootNodeFromAnchor(idl);
 
-    visit(
+    const projectName = visit(
         node,
-        renderVisitor(path.join(__dirname, project, 'src', 'generated'), {
+        renderVisitor(path.join(__dirname, project, 'parser'), {
+            cargoAdditionalDependencies: [
+                `codama-renderers-rust-e2e-${project} = { path = "../../../../renderers-rust/e2e/${project}" }`,
+            ],
             sdkName: sdkName,
-            crateFolder: path.join(__dirname, project),
+            project: `codama-renderers-vixen-parser-e2e-${project}`,
+            generateProto: generateProto,
+            crateFolder: path.join(__dirname, project, 'parser'),
```

Comment:
We are adding a lot of vixen specific params. Overall like to avoid this so lets discuss.